homeassistant/components/image_processing/facebox_face_detect.py

```diff
@@ -0,0 +1,105 @@
+"""
+Component that will perform facial detection via a local facebox instance.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/image_processing.facebox_face_detect
+"""
+import base64
+import requests
+import logging
+import time
+import voluptuous as vol
+
+from homeassistant.core import split_entity_id
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.image_processing import (
+    PLATFORM_SCHEMA, CONF_SOURCE, CONF_ENTITY_ID,
+    CONF_NAME)
+from homeassistant.const import (CONF_IP_ADDRESS, CONF_PORT)
+from homeassistant.components.image_processing.microsoft_face_identify import (
+    ImageProcessingFaceEntity)
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_IP_ADDRESS): cv.string,
+    vol.Required(CONF_PORT): cv.string,
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the classifier."""
+    entities = []
+    for camera in config[CONF_SOURCE]:
+        entities.append(FaceboxFaceDetectEntity(
+            config[CONF_IP_ADDRESS],
+            config[CONF_PORT],
+            camera[CONF_ENTITY_ID],
+            camera.get(CONF_NAME)
+        ))
+    add_devices(entities)
+
+
+class FaceboxFaceDetectEntity(ImageProcessingFaceEntity):
+    """Perform a classification via a Facebox."""
+
+    def __init__(self, ip, port, camera_entity, name=None):
+        """Init with the API key and model id"""
+        super().__init__()
+        self._url = "http://{}:{}/facebox/check".format(ip, port)
+        self._camera = camera_entity
+        if name:
+            self._name = name
+        else:
+            self._name = "Facebox {0}".format(
+                split_entity_id(camera_entity)[1])
+        self._response_time = None
+        self.total_faces = 0
+        self.faces = []
+
+    def process_image(self, image):
+        """Process an image."""
+        timer_start = time.perf_counter()
+        response = {}
+        try:
+            response = requests.post(
+                self._url,
+                json=self.encode_image(image),
+                timeout=30
+                ).json()
+        except requests.exceptions.ConnectionError:
+            _LOGGER.error("ConnectionError: Is Facebox running?")
+            response['success'] = False
+
+        if response['success']:
+            elapsed_time = time.perf_counter() - timer_start
+            self._response_time = "{} seconds".format(
```

Comment:
Please update the time only if you find things. Otherwise it flush the state database.

Reply:
you mean if `total_faces` >  zero?

Reply:
yes. If there is no faces found, he should also not update time.

Reply:
@pvizeli That would mean that the `response_time` was for an earlier (successful request), and not for the last successful request (that returned zero faces). I think the confusion that could cause outweighs any benefit from not updating? The only alternative I can think of is to drop `response_time`  but it is useful for choosing a `scan_interval` and monitoring performance of the classifier under different loads